core/pkg/launch/config_files.go

```diff
@@ -0,0 +1,58 @@
+package launch
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/wandb/wandb/core/internal/data_types"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+// Represents a config file parameter for a job.
+//
+// The relpath is the path to the config file relative to the files directory.
+// The includePaths and excludePaths are used to filter down the run config
+// before it is converted to a schema and saved as job inputs.
+type configFileParameter struct {
+	relpath      string
+	includePaths []ConfigPath
+	excludePaths []ConfigPath
+}
+
+func newFileInputFromRequest(
+	request *service.JobInputRequest,
+) (*configFileParameter, error) {
+	includePaths := make([]ConfigPath, len(request.GetIncludePaths()))
+	excludePaths := make([]ConfigPath, len(request.GetExcludePaths()))
+	for i, path := range request.GetIncludePaths() {
+		includePaths[i] = ConfigPath(path.Path)
+	}
+	for i, path := range request.GetExcludePaths() {
+		excludePaths[i] = ConfigPath(path.Path)
+	}
+	switch source := request.GetSource().GetSource().(type) {
+	case *service.JobInputSource_File:
+		return &configFileParameter{
+			source.File.GetPath(), includePaths, excludePaths}, nil
+	default:
+		return nil, fmt.Errorf("jobBuilder: invalid source type for file input")
+	}
+}
+
+// Infers the structure of a config file.
+//
+// This returns the tree structure and data types of the given config file after filtering
+// its subtrees according to the 'include' and 'exclude' paths in the record.
+func (j *JobBuilder) generateConfigFileSchema(
+	configFile *configFileParameter,
+) data_types.TypeRepresentation {
+	path := filepath.Join(j.settings.FilesDir.GetValue(), "configs", configFile.relpath)
```

Comment:
should this be a user configureable setting? or you want it to be hard-coded to configs?

Reply:
This should not be user customizable, it's an implementation detail. Of course users will be able to see this dir/these files in the files tab but we are already exposing a bunch of internals there